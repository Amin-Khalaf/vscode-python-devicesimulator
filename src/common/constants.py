from enum import Enum

MAC_OS = "darwin"

TIME_DELAY = 0.03

ERROR_SENDING_EVENT = "Error trying to send event to the process : "
DEFAULT_PORT = "5577"

ACTIVE_DEVICE_FIELD = "active_device"
