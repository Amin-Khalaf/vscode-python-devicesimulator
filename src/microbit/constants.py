INDEX_ERR = "index out of bounds"
BRIGHTNESS_ERR = "brightness out of bounds"

LED_WIDTH = 5
LED_HEIGHT = 5

NOT_IMPLEMENTED_ERROR = "This method is not implemented by the simulator"

BOAT = (
    [0, 5, 0, 5, 0],
    [0, 5, 0, 5, 0],
    [0, 5, 0, 5, 0],
    [9, 9, 9, 9, 9],
    [0, 9, 9, 9, 0],
)

HEART = [
    [0, 9, 0, 9, 0],
    [9, 9, 9, 9, 9],
    [9, 9, 9, 9, 9],
    [0, 9, 9, 9, 0],
    [0, 0, 9, 0, 0],
]

BLANK = [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
]


COPY_ERR_MESSAGE = "please copy() first"

LED_MAX = 5
