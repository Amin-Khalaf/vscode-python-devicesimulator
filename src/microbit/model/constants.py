BLANK = "00000:00000:00000:00000:00000:"
BOAT = "05050:05050:05050:99999:09990:"
BRIGHTNESS_ERR = "brightness out of bounds"
COPY_ERR_MESSAGE = "please copy() first"

HEART = "09090:99999:99999:09990:00900:"
INCORR_IMAGE_SIZE = "image data is incorrect size"
INDEX_ERR = "index out of bounds"

LED_HEIGHT = 5
LED_MAX = 5
LED_WIDTH = 5

NOT_IMPLEMENTED_ERROR = "This method is not implemented by the simulator"
UNSUPPORTED_ADD_TYPE = "unsupported types for __add__:"
SAME_SIZE_ERR = "images must be the same size"
