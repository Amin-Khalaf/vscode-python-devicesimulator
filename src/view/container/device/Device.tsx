// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Pivot, PivotItem, PivotLinkFormat } from "office-ui-fabric-react";
import * as React from "react";
import Simulator from "../../components/Simulator";
import { TOOLBAR_ICON_ID } from "../../components/toolbar/SensorModalUtils";
import ToolBar from "../../components/toolbar/ToolBar";
import * as TOOLBAR_SVG from "../../svgs/toolbar_svg";

const CPX_TOOLBAR_BUTTONS: Array<{ label: any; image: any }> = [
    {
        image: TOOLBAR_SVG.SLIDER_SWITCH_SVG,
        label: TOOLBAR_ICON_ID.SWITCH,
    },
    {
        image: TOOLBAR_SVG.PUSH_BUTTON_SVG,
        label: TOOLBAR_ICON_ID.PUSH_BUTTON,
    },
    {
        image: TOOLBAR_SVG.RED_LED_SVG,
        label: TOOLBAR_ICON_ID.RED_LED,
    },
    {
        image: TOOLBAR_SVG.SOUND_SVG,
        label: TOOLBAR_ICON_ID.SOUND,
    },
    {
        image: TOOLBAR_SVG.TEMPERATURE_SVG,
        label: TOOLBAR_ICON_ID.TEMPERATURE,
    },
    {
        image: TOOLBAR_SVG.LIGHT_SVG,
        label: TOOLBAR_ICON_ID.LIGHT,
    },
    {
        image: TOOLBAR_SVG.NEO_PIXEL_SVG,
        label: TOOLBAR_ICON_ID.NEO_PIXEL,
    },
    {
        image: TOOLBAR_SVG.SPEAKER_SVG,
        label: TOOLBAR_ICON_ID.SPEAKER,
    },
    {
        image: TOOLBAR_SVG.MOTION_SVG,
        label: TOOLBAR_ICON_ID.MOTION,
    },
    {
        image: TOOLBAR_SVG.IR_SVG,
        label: TOOLBAR_ICON_ID.IR,
    },
    {
        image: TOOLBAR_SVG.GPIO_SVG,
        label: TOOLBAR_ICON_ID.GPIO,
    },
];
interface IState {
    currentDevice: string | undefined;
}
interface IProps {
    children?: any;
}
const DEVICE_LIST_KEY = {
    CPX: "cpx",
    MICROBIT: "microbit",
};
// Container to switch between multiple devices

class Device extends React.Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            currentDevice: DEVICE_LIST_KEY.MICROBIT,
        };
    }
    render() {
        return (
            <div className="device-container">
                <Pivot
                    linkFormat={PivotLinkFormat.tabs}
                    onLinkClick={this.handleTabClick}
                >
                    <PivotItem
                        headerText="Micro:bit"
                        itemKey={DEVICE_LIST_KEY.MICROBIT}
                    />

                    <PivotItem headerText="CPX" itemKey={DEVICE_LIST_KEY.CPX} />
                </Pivot>
                <div>
                    <div
                        style={{
                            visibility:
                                this.state.currentDevice ===
                                DEVICE_LIST_KEY.MICROBIT
                                    ? "visible"
                                    : "hidden",
                        }}
                    >
                        microbit
                    </div>
                    <div
                        style={{
                            visibility:
                                this.state.currentDevice === DEVICE_LIST_KEY.CPX
                                    ? "visible"
                                    : "hidden",
                        }}
                    >
                        <Simulator />
                        <ToolBar buttonList={CPX_TOOLBAR_BUTTONS} />
                    </div>
                </div>
            </div>
        );
    }
    private handleTabClick = (item: PivotItem | undefined): void => {
        if (item) {
            this.setState({ currentDevice: item.props.itemKey });
            console.log(this.state.currentDevice);
        }
    };
}
export default Device;
