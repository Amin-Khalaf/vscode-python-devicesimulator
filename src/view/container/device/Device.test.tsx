import * as React from "react";
import * as ReactDOM from "react-dom";
import { IntlProvider } from "react-intl";
import * as testRenderer from "react-test-renderer";
import { DEVICE_LIST_KEY } from "../../constants";
import { Device } from "./Device";

describe("Device component", () => {
    it("renders correctly", () => {
        const component = testRenderer
            .create(
                <IntlProvider locale="en">
                    <Device currentSelectedDevice={DEVICE_LIST_KEY.MICROBIT} />
                </IntlProvider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <IntlProvider locale="en">
                <Device currentSelectedDevice={DEVICE_LIST_KEY.MICROBIT} />
            </IntlProvider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});
