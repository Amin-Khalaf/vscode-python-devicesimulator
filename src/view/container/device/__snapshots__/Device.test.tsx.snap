// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device component renders correctly 1`] = `
<div
  className="device-container"
>
  <p>
    Microbit to be implemented!
  </p>
</div>
`;
